File: frontend/src/services/metaDataApi.js
import axios from 'axios';
import { API_BASE_URL } from '../config/env';

const API_BASE = String(API_BASE_URL || '').replace(/\/+$/, '');

export const getTrailerTypes = async () => {
  const res = await axios.get(`${API_BASE}/meta/trailer_types`);
  return res.data;
};

export const getShipmentTypes = async () => {
  const res = await axios.get(`${API_BASE}/meta/shipment_types`);
  return res.data;
};

export const getLocations = async () => {
  const res = await axios.get(`${API_BASE}/meta/locations`);
  return res.data;
};

File: backend/routes/meta_data_api.py
from fastapi import APIRouter
from backend.data.trailer_types import TRAILER_TYPES
from backend.data.canada_us_locations import CANADA_US_LOCATIONS

router = APIRouter(prefix="/api/v1/meta", tags=["meta"])

@router.get("/trailer-types")
def get_trailer_types():
    return {"trailer_types": TRAILER_TYPES}

@router.get("/locations")
def get_locations():
    return {"locations": CANADA_US_LOCATIONS}
